### PR TITLE
Stats: Put 30-days summary behind paywall

### PR DESCRIPTION
**Pull request by a8ck3n**
Related to: https://github.com/Automattic/red-team/issues/317

## Proposed Changes

Adds the "30 days" button in the Summary page navigation to the list of paywalled features. This means the button will have the lock icon and clicking it will throw up the upsell card instead of updating the routing.

SCR-20241212-ojqd

## Why are these changes being made?

For consistency with other locked ranges.

## Testing Instructions

- Open the Calypso Live site.
- Navigate to a new site without a paid plan.
- Manually update the URL to a summary page: `/stats/day/referrers/SITE_SLUG`
- Confirm the "30 Days" button is locked.

Will be gated for all sites after 2024-01-09 without a paid plan.

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
    - [ ] For UI changes, have we tested the change in various languages (for example, ES, PT, FR, or DE)? The length of text and words vary significantly between languages. 
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

File: client/my-sites/stats/hooks/use-should-gate-stats.ts [hunk 1]
```diff
@@ -115,6 +116,7 @@ const gatedStats = [
 	STATS_FEATURE_INTERVAL_DROPDOWN_WEEK,
 	STATS_FEATURE_INTERVAL_DROPDOWN_MONTH,
 	STATS_FEATURE_INTERVAL_DROPDOWN_YEAR,
+	STATS_FEATURE_SUMMARY_LINKS_30_DAYS,
```
Comment (dognose24) [inline client/my-sites/stats/hooks/use-should-gate-stats.ts:121]:
```suggestion
	STATS_FEATURE_SUMMARY_LINKS_7_DAYS,
	STATS_FEATURE_SUMMARY_LINKS_30_DAYS,
```
We may also need to gate even `7 days` as the whole Stats modules are all gated on the Traffic page for Free sites created after the `stats/paid-wpcom-v3` release.

截圖 2024-12-12 晚上10 39 34

File: client/my-sites/stats/hooks/use-should-gate-stats.ts [hunk 2]
```diff
@@ -148,6 +150,7 @@ const basicStats = [
...
```
Comment (dognose24) [inline client/my-sites/stats/hooks/use-should-gate-stats.ts]: Should we keep the `30 days` option usable as it was for sites created before the `stats/paid-wpcom-v3` release?

Comment (lsl) [reply client/my-sites/stats/hooks/use-should-gate-stats.ts]: Related thread? p1734000338014269-slack-C0438NHCLSY / https://github.com/Automattic/wp-calypso/pull/97431

Comment (lsl) [reply client/my-sites/stats/hooks/use-should-gate-stats.ts]: Yes, I think so.

```suggestion
```

File: client/my-sites/stats/hooks/use-should-gate-stats.ts (hunk 1 above)
Comment (lsl) [reply client/my-sites/stats/hooks/use-should-gate-stats.ts:121]: Makes sense to me, 

v2
Screenshot 2024-12-13 at 12-47-38 Jetpack Stats ‹ Site Title — WordPress com

v3
Screenshot 2024-12-13 at 12-48-03 Site Title — WordPress com

**Review by lsl — APPROVED** [review]
LGTM

**Pull request merged**